lib/msf/core/payload/adapter/fetch/server/https.rb

```diff
@@ -0,0 +1,144 @@
+        Msf::OptBool.new('FETCH_CHECK_CERT', [true,"Check SSL certificate", false])
+
+      ]
+    )
+    register_advanced_options(
+      [
+        Msf::OptString.new('FetchHttpServerName', [true, 'Http Server Name', 'Apache']),
+        Msf::OptPath.new('FetchSSLCert', [ false, 'Path to a custom SSL certificate (default is randomly generated)', '']),
+        Msf::OptBool.new('FetchSSLCompression', [ false, 'Enable SSL/TLS-level compression', false ]),
+        Msf::OptString.new('FetchSSLCipher', [ false, 'String for SSL cipher spec - "DHE-RSA-AES256-SHA" or "ADH"']),
+        Msf::OptEnum.new('FetchSSLVersion',
+                         'Specify the version of SSL/TLS to be used (Auto, TLS and SSL23 are auto-negotiate)',
+                         enums: Rex::Socket::SslTcp.supported_ssl_methods)
+      ]
+    )
+  end
+
+  def add_resource(fetch_service, uri)
+    if fetch_service.resources.include?(uri)
+      fail_with(Msf::Exploit::Failure::BadConfig, "Resource collision detected.  Set FETCH_URI to a different value to continue.")
+    end
+    fetch_service.add_resource(uri,
+                               'Proc' => proc do |cli, req|
+                                 on_request_uri(cli, req, srvexe)
+                               end,
+                               'VirtualDirectory' => true)
+  rescue  ::Exception => e
+    fetch_service.deref
+    cleanup_handler
+    fail_with(Msf::Exploit::Failure::Unknown, "Failed to add resource\n #{e}")
+  end
+
+  def cleanup_http_fetch_service(fetch_service)
+    unless fetch_service.nil?
+      fetch_service.remove_resource('/' + srvuri) unless fetch_service.nil?
+      if fetch_service.resources.empty?
+        # if we don't call deref, we cannot start another httpserver
+        # this is a reimplementation of the cleanup_service method
+        # in Exploit::Remote::SocketServer
+        temp_service = fetch_service
+        fetch_service = nil
+        temp_service.cleanup
+        temp_service.deref
+      end
+    end
+  end
+
+  def fetch_protocol
+    'HTTPS'
+  end
+
+  def on_request_uri(cli, request, srvexe)
+    client = cli.peerhost
+    print_status("Client #{client} requested #{request.uri}")
+    if (user_agent = request.headers['User-Agent'])
+      client += " (#{user_agent})"
+    end
+    vprint_status("Client #{client} requested #{request.uri}")
+    vprint_status("Sending payload to #{client}")
+    cli.send_response(payload_response(srvexe))
+  end
+
+  def payload_response(srvexe)
+    res = Rex::Proto::Http::Response.new(200, 'OK', Rex::Proto::Http::DefaultProtocol)
+    res['Content-Type'] = 'text/html'
+    res.body = srvexe.to_s.unpack('C*').pack('C*')
+    res
+  end
+
+  def ssl_cert
+    datastore['FetchSSLCert']
+  end
+
+  def ssl_compression
+    datastore['FetchSSLCompression']
+  end
+
+  def ssl_cipher
+    datastore['FetchSSLCipher']
+  end
+
+  def ssl_version
+    datastore['FetchSSLVersion']
+  end
+
+  def start_http_fetch_handler(srvname, srvexe)
```

Comment:
I think `srvexe` is not used here.
```suggestion
  def start_http_fetch_handler(srvname)
```